src/rslice.rs

```diff
@@ -61,35 +61,39 @@ pub struct rustls_slice_slice_bytes<'a> {
     pub(crate) inner: &'a [&'a [u8]],
 }
 
-/// Return the length of the outer slice. If the input pointer is NULL,
-/// returns 0.
+/// Return the length of the outer slice.
+///
+/// # Safety
+///
+/// If the input pointer is NULL, returns 0.
 #[no_mangle]
-pub extern "C" fn rustls_slice_slice_bytes_len(input: *const rustls_slice_slice_bytes) -> size_t {
-    unsafe {
-        match input.as_ref() {
-            Some(c) => c.inner.len(),
-            None => 0,
-        }
+pub unsafe extern "C" fn rustls_slice_slice_bytes_len(
```

Comment:
What was the clippy output that this fixed? So far we haven't marked our function signatures unsafe, even though they all contain an `unsafe` block.

Reply:
```
error: this public function might dereference a raw pointer but is not marked `unsafe`
  --> src/rslice.rs:69:15
   |
69 |         match input.as_ref() {
   |               ^^^^^
   |
   = note: `#[deny(clippy::not_unsafe_ptr_arg_deref)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref
```

This seems spurious - all of our functions dereference raw pointers. It's the nature of what we're working on. I recommend reverting the changes in this file and putting `#[allow(clippy::not_unsafe_ptr_arg_deref)]` at the top of lib.rs.